### PR TITLE
Ho list page

### LINKED ISSUE #32: 16. As a user, I want the List Page to look professional.
### Acceptance Criteria
- [x] Remove "Hello from the /list page!"
- [x] Make everything centered
- [x] Import checkbox, button, and search input components from MUI
- [x] Whenever an item is checked, the item is crossed out on the UI, and the purchase urgency is removed for 24 hrs
- [x] Space out the item and the delete button
- [x] Change the delete button to an icon
- [x] Add route protection for relevant pages 
- [x] Show selected list across all pages

### PR DESCRIPTION
**Pull request by IamHenryOkeke**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

This code change/update updates the List Page UI and provides route protection to relevant pages

## Related Issue
closes #32 

## Acceptance Criteria

 - Remove "Hello from the /list page!"
 - Make everything centered
 - Import checkbox, button, and search input components from MUI
 - Whenever an item is checked, the item is crossed out on the UI, and the purchase urgency is removed for 24 hrs
 - Space out the item and the delete button
 - Change the delete button to an icon
 - Add route protection for relevant pages
 - Show selected list across all pages

## Type of Changes
`Bug fix`
`UI enhancement`

### Before
Screenshot from 2024-03-27 14-29-59

### After
Screenshot from 2024-03-27 14-30-19

## Testing Steps / QA Criteria

Do a git pull and checkout to ho-listPage branch. Run npm start to start development server. Navigate to list page to see UI updates. Also sign out of the app and try to route to the list and mange-list page. You shouldn't be able to access the pages if not signed in

### CODE REVIEW

**Review by ijayhub — APPROVED** [review]
ProtectedRoute working...

**Review by jmahamed — CHANGES_REQUESTED** [review]
Looks really good, i just think the current list name being rendered at the top should have a bold and bigger font

File: src/AuthContext.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	const [currentUser, setCurrentUser] = useState(null);
+	const [pending, setPending] = useState(true);
+
+	useEffect(() => {
+		onAuthStateChanged(auth, (user) => {
+			setCurrentUser(user);
+			setPending(false);
+		});
+	}, []);
+
```
Comment (redapy) [inline src/AuthContext.jsx]: nice use of React Context 💫. We already have this code, for fetching and detecting user changes, in the `useAuth` custom hook so this is a duplication. I would recommend to move the pending state/logic to the `useAuth` hook and use the hook here.

**Review by redapy — APPROVED** [review]
Everything is working as expected @IamHenryOkeke , amazing work! 🚀 
I left a non-blocking comment

**Review by jmahamed — APPROVED** [review]

File: src/AuthContext.jsx (hunk 1 above)
Comment (IamHenryOkeke) [reply src/AuthContext.jsx]: Thank you Reda for pointing that out. I took a closer look at the code and i figured that the using a context wasn't necessary so I removed the context as a whole. The route protection works just fine regardless. Please review

**Review by redapy — APPROVED** [review]
LGTM!

**Pull request merged**